src/App.jsx

```diff
@@ -1,20 +1,32 @@
-import React from 'react';
+import React from "react";
 import TodoList from "./TodoList";
-import TodoForm from "./AddTodoForm"
+import TodoForm from "./AddTodoForm";
+
+const useSemiPersistentState = () => {
+  const [todoList, setTodoList] = React.useState(
+    JSON.parse(localStorage.getItem("savedTodoList")) || []
+  );
+
+  React.useEffect(() => {
+    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
+  }, [todoList]);
+
+  return [todoList, setTodoList];
+};
 
 function App() {
-  const [todoList, setTodoList] = React.useState([])
+  const [todoList, setTodoList] = useSemiPersistentState("");
```

Comment:
`useSemiPersistentState` doesn't take any arguments

Reply:
so should I just remove change ("") to ()